crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -166,8 +166,12 @@ impl OwnUserIdentity {
             .store
             .get_user_devices(self.user_id())
             .await?
-            .into_keys()
-            .filter(|d| &**d != self.verification_machine.own_device_id())
+            .into_iter()
+            .filter_map(|(d_id, d)| {
+                (d_id != self.verification_machine.own_device_id()
```

Comment:
Let's move this into a separate method, that way this also becomes testable. I don't require a test but would be nice.

Reply:
What would the signature look like?

1. `fn should_send_request_to_device(&self, device_id: &DeviceId, own_device_id: &DeviceId, device: &ReadOnlyDevice) -> bool`
2. `fn devices_to_request(&self, devices: HashMap<OwnedDeviceId, ReadOnlyDevice>, own_device_id: &DeviceId) -> Vec<OwnedDeviceId>`

Or something else even?

Reply:
I think `UserDevices::signed_devices(&self) -> Iterator<Item = Device>` would be the most pleasant way to go about this.

Reply:
We don't have access to `UserDevices` here. It's a type returned by the `Store`, we only have access to the `VerificationStore` that wraps a `dyn CryptoStore` that returns a `HashMap<OwnedDeviceId, ReadOnlyDevice>`, afaict.

Reply:
Ah right, that's a shame. I guess option 2. from your proposal then sounds fine.